### PR TITLE
[CLD-4568] Create installation SLIs based on ring preferences

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Create installation SLIs based on ring preferences

#### Release Note
```release-note
- Create installation SLIs based on ring preferences
```

### CODE REVIEW

**Review by stafot — APPROVED** [review]
LGTM

File: internal/provisioner/cluster_installation_provisioner.go [hunk 1]
```diff
@@ -312,9 +314,18 @@ func (provisioner *CommonProvisioner) updateClusterInstallation(
...
+		logger.Debug("Creating or updating Mattermost installation SLI")
+		err = createOrUpdateInstallationSLI(clusterInstallation, k8sClient, logger)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create cluster installation SLI %s", clusterInstallation.ID)
+		}
+	} else {
+		logger.Debug("Removing Mattermost installation SLI as installation group not in the approved list")
+		err = deleteInstallationSLI(clusterInstallation, k8sClient, logger)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create cluster installation SLI %s", clusterInstallation.ID)
```
Comment (fmartingr) [inline internal/provisioner/cluster_installation_provisioner.go:327]:
```suggestion
			return errors.Wrapf(err, "failed to delete cluster installation SLI %s", clusterInstallation.ID)
```

**Review by fmartingr — APPROVED** [review]
As long as the delete logic is correctly put (since I see that is added but was not present before) this LGTM.

**Review by gabrieljackson — APPROVED** [review]
LGTM!

**Pull request merged**